File: RobWork/example/python/ex_frame_to_frame_transforms.py
from rw import *

def frameToFrameTransforms(a, b, tree_structure, states):
    fk = FKRange(a, b, tree_structure);
    
    result = Transform3dVector();
    for state in states:
        result.push_back(fk.get(state));
    return result;

File: RobWork/example/java/src/ExFrameToFrameTransforms.java
import java.util.Iterator;

import org.robwork.rw.*;

public class ExFrameToFrameTransforms {
    public static Transform3dVector frameToFrameTransforms(
            Frame a, Frame b, State tree_structure, StateVector states)
    {
        FKRange fk = new FKRange(a, b, tree_structure);

        Transform3dVector result = new Transform3dVector();
        Iterator<State> iterator = states.iterator();
        while(iterator.hasNext()) {
            State state = iterator.next();
            result.add(fk.get(state));
        }
        return result;
    }
}